mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeansModel.scala

```diff
@@ -37,7 +37,7 @@ import org.apache.spark.sql.{Row, SparkSession}
  */
 @Since("0.8.0")
 class KMeansModel @Since("2.4.0") (@Since("1.0.0") val clusterCenters: Array[Vector],
-  @Since("2.4.0") val distanceMeasure: String)
+  @Since("2.4.0") val distanceMeasure: String, @Since("2.4.0") val trainingCost: Double)
```

Comment:
Since we changed the constructor here, and since it is not private, we should provide a similar (and deprecated) constructor without training cost which calls this with the default value.

Reply:
This constructor was introduced by a previous PR for 2.4, so this was never out (therefore I think we don't need to keep it).

Reply:
That's awesome, lets try and get that fixed up before 2.4 goes so we don't have to any workarounds :)